[CHANGELOG] v4.0.0-alpha.34 (#2180)
mui-org/material-ui-x: CHANGELOG.md

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [4.0.0-alpha.34](https://github.com/mui-org/material-ui-x/compare/v4.0.0-alpha.33...v4.0.0-alpha.34)

_July 21, 2021_

Big thanks to the 11 contributors who made this release possible. Here are some highlights ✨:

> 

Reviewer: The most important change of this release is IMHO that we fix v4 support. Could we mention it?

Author: Sure

- 🐞 Add \"is empty\" and \"is not empty\" operators (#1997) @m4theushw
- 💅 Improve the editing API (#1955) @m4theushw

> - 🐛 We have fixed the `scrollToIndexes` behavior (#2162) @oliviertassinari

Reviewer: It's not great yet, it's still completely broken when we scroll enough done, and it's desorienting when going backwards.

```suggestion
- 🐛 We have improved the scroll keyboard (#2162) @oliviertassinari
```

- ⚡️ Add control state for selection model, filter model, and sort model @dtassone
- 💡 Add quick filter demo in the docs @dtassone

### @material-ui/x-grid@v4.0.0-alpha.34 / @material-ui/data-grid@v4.0.0-alpha.34

#### Breaking changes

- [DataGrid] Fix scrollToIndexes behavior (#2162) @oliviertassinari

  Remove public `apiRef.current.isColumnVisibleInWindow()` as it servers private use cases.

  ```diff
  -apiRef.current.isColumnVisibleInWindow()
  ```

- [DataGrid] Remove stateId argument from GridApi getState method (#2141) @flaviendelangle

  ```diff
  -const filterState = apiRef.current.getState('filter');
  +const filterState = apiRef.current.getState().filter;
  ```

- [DataGrid] Improve controllable sorting (#2095) @dtassone

  ```diff
   // Props
  -onSortModelChange?: (params: GridSortModelParams) => void;
  +onSortModelChange?: (model: GridSortModel) => void;

   // Usage
  -<DataGrid onSortModelChange = {(params: GridSortModelParams)=>  setSortModel(params.model)} />
  +<DataGrid onSortModelChange = {(model: GridSortModel)=> setSortModel(model) } />
  ```

- [DataGrid] Improve controllable filter (#1909) @dtassone

>   ```diff
>    // Props
>   -onFilterModelChange?: (params: GridFilterModelParams) => void;
>   +onFilterModelChange?: (model: GridFilterModel) => void;
> 

Reviewer 2: ```suggestion
  Update the prop arguments:
  ```diff
  -onFilterModelChange?: (params: GridFilterModelParams) => void;
  +onFilterModelChange?: (model: GridFilterModel) => void;
  ```
```

>    // Usage
>   -<DataGrid onFilterModelChange = {(params: GridFilterModelParams)=>  setFilterModel(params.model)} />
>   +<DataGrid onFilterModelChange = {(model: GridFilterModel)=> setFilterModel(model) } />

Reviewer 2: ```suggestion
  Update the model with the first argument directly:
  ```diff
  -<DataGrid onFilterModelChange = {(params: GridFilterModelParams)=>  setFilterModel(params.model)} />
  +<DataGrid onFilterModelChange = {(model: GridFilterModel)=> setFilterModel(model) } />
```

  ```

- [DataGrid] Improve the editing API (#1955) @m4theushw

>   -  The `props` key in the first argument of `commitCellChange` was removed to promote the use of the value already stored in the state. To update the value in the state, call `setEditCellProps` before.

Reviewer 2: ```suggestion
  - The `props` key in the first argument of `commitCellChange` was removed to promote the use of the value already stored in the state. 
    To update the value in the state, call `setEditCellProps` before.
```

    ```diff
    +apiRef.current.setCellMode(1, 'name', 'edit');
    apiRef.current.commitCellChange({ id: 1, field: 'name' });
    ```

  - The `setCellValue` was removed from the API. Use `commitCellChange` or `updateRows` in place.

    ```diff
    -apiRef.current.setCellValue({ id: 1, field: 'name', value: 'Ana' });
    +apiRef.current.updateRows([{ id: 1, name: 'Ana' }]);
    ```

    or

    ```diff
    -apiRef.current.setCellValue({ id: 1, field: 'name', value: 'Ana' });
    +apiRef.current.setCellMode(1, 'name', 'edit');
    +apiRef.current.setEditCellProps({ id: 1, field: 'name', props: { value: 'Ana' } });
    +apiRef.current.commitCellChange({ id: 1, field: 'name' });
    ```

  - [DataGrid] The `getEditCellProps` was removed because `getEditCellPropsParams` offers the same functionality.

    ```diff
    -const props = apiRef.current.getEditCellProps(1, 'name');
    +const { props } = apiRef.current.getEditCellPropsParams(1, 'name');
    ```

    **Note**: This method will now throw an error if the cell is in view mode.

- [DataGrid] Implement useControlState hook, and add control state on selectionModel (#1823) @dtassone

>   ```diff

Reviewer 2: Same as before, put the comments outside and split the diffs.

   // Props
  -props.onRowSelected
  -onSelectionModelChange?: (params: GridSelectionModelChangeParams) => void;
  +onSelectionModelChange?: (model: GridSelectionModel) => void;

   // Usage
  -<DataGrid onSelectionModelChange = {(params: GridSelectionModelChangeParams)=>  setSelectionModel(params.model)} />
  +<DataGrid onSelectionModelChange = {(model: GridSelectionModel)=> setSelectionModel(model) } />
  ```

#### Changes

- [DataGrid] Use find instead of filter (#2015) @DanailH
- [DataGrid] Add \"is empty\" and \"is not empty\" operators (#1997) @m4theushw
- [DataGrid] Add `minWidth` to `GridColDef` (#2101) @DanailH
- [DataGrid] Add missing localeText types (#2118) @oliviertassinari

> - [DataGrid] Add missing translations to frFR locale (#2082) @flaviendelangle

Reviewer 2: ```suggestion
- [DataGrid] Add missing translations to French (frFR) locale (#2082) @flaviendelangle
```

- [DataGrid] Add quick filter demo (#2149) @dtassone
- [DataGrid] Allow passing styles and Popper props to GridPanel (#1994) @sebastianfrey
- [DataGrid] Allow to customize the columns exported as CSV (#2008) @flaviendelangle

> - [DataGrid] Emit `GRID_PAGE_SIZE_CHANGE` when autoPageSize is set and the grid size changes (#1986) @flaviendelangle

Reviewer 2: ```suggestion
- [DataGrid] Emit `pageSizeChange` when autoPageSize is set and the grid size changes (#1986) @flaviendelangle
```

- [DataGrid] Fix crash when id has a single-quote (#2033) @rbrishabh
- [DataGrid] Fix localeText type (#2117) @oliviertassinari
- [DataGrid] Fix manual entry in date fields (#2051) @m4theushw
- [DataGrid] Fix scrollToIndexes offscreen column (#1964) @m4theushw
- [DataGrid] Fix support for `@material-ui/core@4.12` (#2108) @DanailH
- [DataGrid] Improve GridToolbarXXX props flexibility (#2157) @tifosiblack
- [DataGrid] Make GridToolbarXXX props overridable (#2084) @tifosiblack
- [DataGrid] Remove 'hide: true' from a column should correctly resize the others column (#2034) @flaviendelangle
- [DataGrid] Remove focus on cell when its row is removed from the data (#1995) @flaviendelangle
- [DataGrid] Remove unused `editMode` prop (#2173) @ZeeshanTamboli
- [DataGrid] Support style prop (#2116) @oliviertassinari
- [DataGrid] Use Intl.Collator for string comparison (#2155) @m4theushw

> - [DataGrid] apiRef.current.getRow can return null (#2010) @flaviendelangle

Reviewer 2: ```suggestion
- [DataGrid] Update apiRef.current.getRow to signal that it can return a null value (#2010) @flaviendelangle
```
